### PR TITLE
Decompile func_800FDD44

### PR DESCRIPTION
**Pull request by sozud**

### CODE REVIEW

File: src/dra/42398.c [hunk 1]
```diff
@@ -3059,7 +3059,26 @@ bool func_800FDC94(s32 arg0) {
...
+            if (temp_v0 == 0) {
+                player_equip_head[arg0] = 0;
+                func_800F53A4();
+                return 1;
+            }
+            D_8009798A[temp_s0]--;
+        }
+    }
+    return 0;
+}
```
Comment (sonicdcer) [inline src/dra/42398.c:3081]:
```suggestion
bool func_800FDD44(s32 equipHeadIndex) {
    s32 equipedItem;
    u8 temp_s1;
    u8 temp_v0;

    equipedItem = player_equip_head[equipHeadIndex];
    temp_s1 = D_800A4B1D[player_equip_head[equipHeadIndex]].unk0;
    if (!(func_800FD7C0(0x54, 4))) {
        if (temp_s1 != 0) {
            temp_v0 = D_8009798A[equipedItem];
            if (temp_v0 == 0) {
                player_equip_head[equipHeadIndex] = 0;
                func_800F53A4();
                return true;
            }
            D_8009798A[equipedItem]--;
        }
    }
    return false;
}
```

File: include/dra.h [hunk 2]
```diff
@@ -932,6 +932,7 @@ extern const char* c_strTime;
 extern const char* c_strALUCART;
 extern const char* c_strSSword;
 extern s32 D_800A4B04;
+extern s8 D_800A4B1D[];
```
Comment (sonicdcer) [inline include/dra.h:935]:
```suggestion
extern Unkstruct_800A4B12 D_800A4B1D[];
```

**Review by sonicdcer — CHANGES_REQUESTED** [review]
Good work!

**Review by Xeeynamo — APPROVED** [review]
`player_equip_head[equipHeadIndex]` start to look sus. Probably that is that an equipment array of size 5. But so far this looks good!

**Pull request merged**